src/System.Runtime.Analyzers/Core/InstantiateArgumentExceptionsCorrectly.cs

```diff
@@ -1,59 +1,189 @@
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: s_helpUri,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
         internal static DiagnosticDescriptor IncorrectParameterNameRule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessageIncorrectParameterName,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: s_helpUri,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(NoArgumentsRule, IncorrectMessageRule, IncorrectParameterNameRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(
+                compilationContext =>
+                {
+                    Compilation compilation = compilationContext.Compilation;
+                    ITypeSymbol stringType = compilation.GetSpecialType(SpecialType.System_String);
+                    ITypeSymbol argumentExceptionType = compilation.GetTypeByMetadataName("System.ArgumentException");
+
+                    Debug.Assert(stringType != null);
+                    if (argumentExceptionType == null)
+                        return;
```

Comment:
Minor: Roslyn code bases wrap even single line statements in a block with curly braces - we try to follow this in roslyn-analyzers as well.